File: Shared/Resources/gnomesort.bundle/gnomesort.js
function gnomeSort(arr) {
  const n = arr.length;
  let index = 0;
  while (index < n) {
    if (index == 0) {
      index++;
    }
    if (arr[index] >= arr[index - 1]) {
      index++;
    } else {
      let swapTemp = arr[index];
      arr[index] = arr[index - 1];
      arr[index - 1] = swapTemp;
      index--;
    }
  }
}

var array = [35, 95, 74, 71, 72, 30, 96, 53, 9, 0];
gnomeSort(array);
console.log(array);

File: Shared/Resources/gnomesort.bundle/gnomesort.cs
using System;

public class GnomeSort {
  public static void Sort(int[] arr) {
    int n = arr.Length;
    int index = 0;
    while (index < n) {
      if (index == 0) {
        index++;
      }
      if (arr[index] >= arr[index - 1]) {
        index++;
      } else {
        (arr[index], arr[index - 1]) = (arr[index - 1], arr[index]);
        index--;
      }
    }
  }

  public static void Main(String[] args) {
    int[] array = {35, 95, 74, 71, 72, 30, 96, 53, 9, 0};
    Sort(array);
    string result = "[" + String.Join(", ", array) + "]";
    Console.WriteLine(result);
  }
}